src/Application.QueryServices/Services/TransactionsQueryService.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace PetProjects.MicroTransactionsApi.Application.QueryServices.Services
+{
+    using System;
+    using System.Threading.Tasks;
+    using PetProjects.Framework.Cqrs.Mediator;
+    using PetProjects.MicroTransactionsApi.Application.Dto;
+    using PetProjects.MicroTransactionsApi.Application.QueryServices.Mappings;
+    using PetProjects.MicroTransactionsApi.Domain.Queries;
+    using PetProjects.MicroTransactionsApi.Domain.Queries.ReadModels;
+
+    public class TransactionsQueryService : ITransactionsQueryService
+    {
+        private readonly ISimpleMediator mediator;
+
+        public TransactionsQueryService(ISimpleMediator mediator)
+        {
+            this.mediator = mediator;
+        }
+
+        public async Task<TransactionByIdDto> GetTransactionByIdAsync(Guid id)
+        {
+            var query = new GetTransactionByIdQuery(id);
```

Comment:
I think we could simplify this, by moving the Mediator to the Controller. Create the query and validate it there (faster feedback in my opinion) and make the query call there. I think it would simplify the structure and call. It's one less class to maintain.

Reply:
It will have access to projections, views from the write model, right?

Reply:
While would a Query have access to the write model?

Reply:
This comment was misplaced, I mean the application itself won't have access to the write model.

Of course, a query should not access the write model.

Reply:
Moved Mediator to Controller in 6810897